app/lwip/app/espconn_tcp.c

```diff
@@ -430,12 +434,14 @@ espconn_Task(os_event_t *events)
 				break;
 			case SIG_ESPCONN_ERRER:
 				/*remove the node from the client's active connection list*/
-				espconn_list_delete(&plink_active, task_msg);
+			    if (espconn_manual_recv_enabled(task_msg))
+			        espconn_list_delete(&plink_active, task_msg);
```

Comment:
This looks like dubious indentation, post-patch.  I believe that the if and list_delete should be 4 spaces further to the right but that the code is plausibly conveying the correct semantics as written?